content/blog/next-level-iac-no-more-clickops/index.md

```diff
@@ -0,0 +1,341 @@
+
+```shell
+$ aws ec2 describe-vpcs --output JSON
+```
+
+This command shows you all the VPCs under your AWS account, in JSON format. Pulumi will let us describe and manage all of that with code, but creating it all from scratch that would need a lot of typing. Luckily, we can import this existing infrastructure with a single command `pulumi import`, which takes a list of resources in JSON format and outputs a Pulumi program in your preferred programming language.
+
+However, the JSON that the AWS CLI outputs isn't in a format that Pulumi understands. Also a VPC isn't made of a single resource, but rather a graph of many types of resources such as *SecurityGroups*, *Subnets*, *RouteTables*, etc. Scanning all the necessary resources and generating Pulumi-compatible JSON is a complicated topic. For that we will need another tool.
+
+## Using Boto to generate Pulumi-compatible JSON
+
+{{% notes type="info" %}}
+The command-line tool `jq` is really handy for viewing and filtering JSON. If you don't already have it set up go ahead and [get that installed](https://jqlang.github.io/jq/download/).
+{{% /notes %}}
+
+We put together a basic demo tool called [`pulumi-import-aws-account-scraper`][pulumi-import-aws-account-scraper] using Python’s [Boto library][boto-lib], which can scrape your AWS account, finding all the various resources. It outputs Pulumi compatible JSON which we can then use to generate a Pulumi program in your preferred language.
+
+First, clone the [`pulumi/pulumi-import-aws-account-scraper`][pulumi-import-aws-account-scraper] repo and setup the tool:
+
+```shell
+git clone git@github.com:pulumi/pulumi-import-aws-account-scraper.git
+cd pulumi-import-aws-account-scraper
+virtualenv -p python3 venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
+Next we will use this demo tool to scan your account:
+
+```shell
+source venv/bin/activate
+python3 account_scraper.py > resources.json
+```
+
+This will generate a JSON file with an entry for each resource under your account. However, we only want to import some of those. For this example, we'll assume there is a VPC named `thor` in that list, with associated resources also containing the term `thor` in their name. We can filter that out with a quick `jq` expression:
+
+```shell
+cat resources.json | jq '.resources |= map(select(.name | contains("thor")))' > resources-filtered.json
+```
+
+If you're working with a different resource type, with different names, feel free to customize this expression to filter out anything you don't want to bring under management by Pulumi.
+
+## Run pulumi import to generate a Pulumi program
+
+Ok, now we've got a file called `resources-filtered.json` in Pulumi-compatible JSON format, describing only the resources we want to import for our particular VPC.
+
+Here's the magic moment! Let’s import that into Pulumi and generate our Pulumi program!
```

Comment:
I think we should double click here and explain what is happening. It is importing the existing VPC into the state store and then generating code for it. Now this state piece if important because if you just execute the code, well it will try to create anew.

Reply:
I also agree here. When I was reading this, I was wondering exactly what was happening that was letting Pulumi know to manage existing resources rather than just create new ones since nothing in the IaC code was visibly indicating that an import was happening.